tests/unit/errors/conftest.py

```diff
@@ -0,0 +1,84 @@
+import json
+from collections import namedtuple
+
+import pytest
+import requests
+
+_TestResponse = namedtuple("_TestResponse", ("data", "r", "method", "url"))
+
+
+def _mk_response(
+    data, status, method=None, url=None, headers=None, data_transform=None
+):
+    resp = requests.Response()
+
+    if data_transform:
+        resp._content = data_transform(data).encode("utf-8")
+    else:
+        resp._content = data.encode("utf-8")
+    resp.encoding = "utf-8"
+
+    if headers:
+        resp.headers.update(headers)
+
+    resp.status_code = str(status)
+    resp.reason = {
+        200: "OK",
+        400: "Bad Request",
+        401: "Unauthorized",
+        403: "Forbidden",
+        404: "Not Found",
+        500: "Server Error",
+    }.get(status, "Unknown")
```

Comment:
This dict [exists in the `http.client` module as `responses`](https://docs.python.org/3/library/http.client.html#http.client.responses):

```python
resp.reason = http.client.responses.get(status, "Unknown")
```

Note, though, that `500` translates to `"Internal Server Error"`.

Reply:
🥳

Truly a joy to replace a quick hack with something from the stdlib. I'll add that change.